### PR TITLE
Any/All method

### PR DESCRIPTION
**Pull request by akleeman**
Adds methods for checking if any/all `bool` values in a vector of `bool` are true.

### CODE REVIEW

File: include/albatross/src/utils/vector_utils.hpp [hunk 1]
```diff
@@ -31,6 +31,30 @@ inline std::vector<double> linspace(double a, double b, std::size_t n) {
   return xs;
 }
 
+inline bool all(const std::vector<bool> &xs) {
```
Comment (jbangelo) [inline include/albatross/src/utils/vector_utils.hpp]: Uh oh, you've stumbled into the dreaded `std::vector<bool>`! There's a specialization in the standard for `std::vector<bool>` which attempts to save space by storing each `bool` in a single bit. The consequence of this is that when you iterate over a `std::vector<bool>` you don't get actual references you get reference like classes.

TL;DR: I think the `for` loops below probably shouldn't make `x` be a reference.

File: include/albatross/src/utils/vector_utils.hpp [hunk 2]
```diff
@@ -31,6 +31,30 @@ inline std::vector<double> linspace(double a, double b, std::size_t n) {
...
+  if (xs.size() == 0) {
+    return false;
+  }
+  for (const auto &x : xs) {
```
Comment (jbangelo) [inline include/albatross/src/utils/vector_utils.hpp]:
```suggestion
  for (const auto x : xs) {
```

File: include/albatross/src/utils/vector_utils.hpp [hunk 3]
```diff
@@ -31,6 +31,30 @@ inline std::vector<double> linspace(double a, double b, std::size_t n) {
   return xs;
 }
 
+inline bool all(const std::vector<bool> &xs) {
+  if (xs.size() == 0) {
+    return false;
```
Comment (jbangelo) [inline include/albatross/src/utils/vector_utils.hpp]: I've usually seen this base case for this all type function default to `true` when empty and the `any` function default to `false` when empty. I guess in this situation you're logic would be simpler with both returning `false`?

Comment (akleeman) [reply include/albatross/src/utils/vector_utils.hpp]: Yeah I wasn't totally sure about the empty cases.  I think I agree that `all` should probably return `true`.  I was tempted to put an assert in there ... but that didn't feel right.
